kernel-core/src/main/java/io/zephyr/kernel/launch/OutputArrayInputStream.java

```diff
@@ -0,0 +1,66 @@
+package io.zephyr.kernel.launch;
+
+import io.zephyr.kernel.misc.SuppressFBWarnings;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import lombok.val;
+
+@SuppressFBWarnings
+@SuppressWarnings("PMD.AvoidUsingVolatile")
+public class OutputArrayInputStream extends InputStream {
+
+  private final List<String> messages;
+  private volatile int readCount;
+  private volatile int currentIdx;
+  private volatile boolean closed;
+
+  public OutputArrayInputStream(List<String> messages) {
+    this.messages = messages;
+    readCount = 0;
+    closed = false;
+    currentIdx = 0;
+  }
+
+  @Override
+  public int read() throws IOException {
+    synchronized (messages) {
+      if (closed) {
+        throw new IOException("Stream closed");
+      }
+      if (readCount < messages.size()) {
```

Comment:
Is the notion that read() gets called again automagically so this can just be an if and not a loop?